File: Esport-Frontend/src/constants/playerPositions.ts
/**
 * Дзеркало Esport-Backend/Common/PlayerPositions.cs.
 *
 * Сервер відхиляє позицію, якої немає в цьому списку, тож форми показують
 * саме його — вільне текстове поле неминуче призводило б до помилки 400.
 * Змінюючи список тут, змініть його і на бекенді.
 */
export const PLAYER_POSITIONS = [
  "Support",
  "ADC",
  "Mid",
  "Jungle",
  "Top",
  "IGL",
  "Entry",
  "Lurker",
  "AWPer",
  "Rifler"
] as const;

export type PlayerPosition = (typeof PLAYER_POSITIONS)[number];

/** Межі, які перевіряє PlayerRules на бекенді. */
export const PLAYER_MIN_AGE = 13;
export const PLAYER_MAX_AGE = 50;
export const NICKNAME_MIN_LENGTH = 3;
export const NICKNAME_MAX_LENGTH = 30;
export const NICKNAME_PATTERN = /^[a-zA-Z0-9_]+$/;

File: Esport-Backend/Common/PlayerPositions.cs
namespace TForge.Common
{
    /// <summary>
    /// Дозволені ігрові позиції. Значення повинні збігатися з тими,
    /// що показує фронтенд у випадному списку — інакше форма надішле
    /// значення, яке валідатор відхилить.
    /// </summary>
    public static class PlayerPositions
    {
        public const string Support = "Support";
        public const string ADC = "ADC";
        public const string Mid = "Mid";
        public const string Jungle = "Jungle";
        public const string Top = "Top";
        public const string IGL = "IGL";
        public const string Entry = "Entry";
        public const string Lurker = "Lurker";
        public const string AWPer = "AWPer";
        public const string Rifler = "Rifler";

        public static readonly string[] All =
        {
            Support, ADC, Mid, Jungle, Top, IGL, Entry, Lurker, AWPer, Rifler
        };

        public static bool IsValid(string? position) => position != null && All.Contains(position);
    }
}
